tests/unit/panther_analysis_tool/test_main.py

```diff
@@ -471,5 +471,26 @@ def test_available_destination_names_valid_name_returned(self):
         return_code, invalid_specs = pat.test_analysis(args)
         assert_equal(return_code, 0)
 
-
-
+    def test_analysis_validation_function(self):
+        import boto3
+        import json
+        from unittest import mock
+        """Validation function only returns analysis that exist in Panther"""
+        requested_deletion = ['Rule.Does.Exist', 'Rule.Doesnt.Exist']
+        args = pat.setup_parser().parse_args(f'delete '
+                                             f'--analysis-id '
+                                             f'{requested_deletion}'.split())
+
+        rv = '{"body":{"paging": {"thisPage": 1, "totalPages": 1, "totalItems": 1}, "detections": [{"complianceStatus": "", "resourceTypes": [], "suppressions": [], "dedupPeriodMinutes": 0, "logTypes": [], "scheduledQueries": [], "summaryAttributes": [], "threshold": 0, "analysisType": "", "body": "", "createdAt": "0001-01-01T00:00:00Z", "createdBy": "", "description": "", "displayName": "", "enabled": false, "id": "Rule.Does.Exist", "lastModified": "0001-01-01T00:00:00Z", "lastModifiedBy": "", "outputIds": [], "packIds": [], "reference": "", "reports": {}, "runbook": "", "severity": "", "tags": [], "tests": [], "versionId": "", "managed": false, "parentId": ""}]}}'
+
+        class MockObject():
```

Comment:
I don't think you need this class. You can set `rv` as a `bytes` multiline literal so that it is also easier to read:

```python
rv = b'''
{
  "body": ...
}
'''
```

And then use a `BytesIO` instance as the return value for the `invoke` method.

```python
from io import BytesIO

invoke_mock.invoke.return_value = BytesIO(rv)
```



I believe this will also resolve [your question above](https://github.com/panther-labs/panther_analysis_tool/pull/162/files#r793115177).

Reply:
Thank you @georgepsarakis , that plus a little bit of manipulation of the payload created a correct mock. Much appreciated